types/wlr_keyboard_group.c

```diff
@@ -0,0 +1,299 @@
+#define _POSIX_C_SOURCE 200809L
+#include <assert.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <time.h>
+#include <wayland-server-core.h>
+#include <xkbcommon/xkbcommon.h>
+#include "wlr/interfaces/wlr_keyboard.h"
+#include "wlr/types/wlr_keyboard.h"
+#include "wlr/types/wlr_keyboard_group.h"
+#include "wlr/util/log.h"
+
+struct keyboard_group_device {
+	struct wlr_keyboard *keyboard;
+	struct wl_listener key;
+	struct wl_listener modifiers;
+	struct wl_listener keymap;
+	struct wl_listener repeat_info;
```

Comment:
We should add a destroy listener in there too.

Reply:
Added. `wlr_keyboard` didn't have a `destroy` signal so I had to add it